### PR TITLE
Add support for plotting PNG images from a compute task.

### PR DESCRIPTION
**Pull request by bsmithyman**

### CODE REVIEW

File: properties_image/tasks.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+from io import BytesIO
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+import properties
+from properties.task import Task, BaseResult
+from .images import ImagePNG
+
+
+class PlotImagePNG(Task):
```
Comment (rowanc1) [inline properties_image/tasks.py]: yeah, this is cool

File: properties_image/tasks.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                'Aspect ratio floating point',
+            ),
+        ),
+        default=1,
+    )
+
+    @property
+    def plt_shape(self):
+        """Shape for matplotlib image result"""
+        return (self.ny, self.nx)
```
Comment (rowanc1) [inline properties_image/tasks.py]: does this work? Where are these defined? Can we put sensible bounds on the size of the image?

File: properties_image/tasks.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    )
+    cmin = properties.Float(
+        'Colorbar minimum',
+        required=False,
+    )
+    cmax = properties.Float(
+        'Colorbar maximum',
+        required=False,
+    )
+    width = properties.Float(
```
Comment (rowanc1) [inline properties_image/tasks.py]: Can you put the units here? are these inches?

File: properties_image/tasks.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        """Image result of computations"""
+
+        image = ImagePNG(
+            'Image plot',
+            required=True,
+        )
+
+    def plot_from_array(self, arr):
+        """Plot a PNG image from an input array"""
+        fig, axes = plt.subplots(1, 1, figsize=(self.width, self.height))
```
Comment (rowanc1) [inline properties_image/tasks.py]: dpi?

File: properties_image/tasks.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            'vmax': self.cmax,
+        }
+        plt.imshow(arr.reshape(self.plt_shape), **plotopts)
+        if self.grid:
+            plt.grid()
+        if self.axes:
+            plt.colorbar()
+            plt.title(self.title)
+        else:
+            plt.axis('off')
```
Comment (rowanc1) [inline properties_image/tasks.py]: does this fill the entire plot?

File: properties_image/tasks.py (hunk 3 above)
Comment (bsmithyman) [inline properties_image/tasks.py]: IIRC, yes, inches. I can look into something standard, but of course it is dependent on DPI too.

File: properties_image/tasks.py (hunk 2 above)
Comment (bsmithyman) [inline properties_image/tasks.py]: Ah, good catch, thanks. Those are in the other half of the task I stripped this out of. I will add something appropriate here.

File: properties_image/tasks.py (hunk 4 above)
Comment (bsmithyman) [inline properties_image/tasks.py]: Yeah, should do.

File: properties_image/tasks.py (hunk 5 above)
Comment (bsmithyman) [inline properties_image/tasks.py]: That combined w/ the window extent bit, yes.

**Pull request merged**